storage/block/types.go

```diff
@@ -137,7 +137,13 @@ type NewDatabaseBlockFn func() DatabaseBlock
 // DatabaseBlock represents a data block.
 type DatabaseBlock interface {
 	OnRetrieveBlock
+	DatabaseBlockPublic
+	// Private methods because only the Wired List itself should use them.
+	databaseBlockPrivate
+}
 
+// DatabaseBlockPublic is the public portion of the DatabaseBlock interface
+type DatabaseBlockPublic interface {
```

Comment:
Yeah instead of adding a `public` and `private` suffix naming convention that we don't use anywhere else, perhaps we can add just have `type databaseBlock interface { /* private methods */ }` and we can just keep the `DatabaseBlockPublic` methods inlined on the `DatabaseBlock` interface.